packages/flutter/lib/src/rendering/sliver_grid.dart

```diff
@@ -596,14 +596,9 @@ class RenderSliverGrid extends RenderSliverMultiBoxAdaptor {
     final int firstIndex = layout.getMinChildIndexForScrollOffset(scrollOffset);
     final int? targetLastIndex = targetEndScrollOffset.isFinite ?
       layout.getMaxChildIndexForScrollOffset(targetEndScrollOffset) : null;
-
     if (firstChild != null) {
-      final int oldFirstIndex = indexOf(firstChild!);
-      final int oldLastIndex = indexOf(lastChild!);
-      final int leadingGarbage = (firstIndex - oldFirstIndex).clamp(0, childCount); // ignore_clamp_double_lint
-      final int trailingGarbage = targetLastIndex == null
-        ? 0
-        : (oldLastIndex - targetLastIndex).clamp(0, childCount); // ignore_clamp_double_lint
+      final int leadingGarbage = _calculateLeadingGarbage(firstIndex);
```

Comment:
I copy over the code from RenderSliverFixedExtentBoxAdaptor

Reply:
Could the logic live in a common place (e.g. the RenderSliverMultiBoxAdaptor base class) and be shared between the two instead of duplicating it?

Reply:
I don't want to expose these method because they are implementation detail. since this are only two instance, I prefer to keep it as is. We can revisit this if there are more use case for these logic